### PR TITLE
Add readiness api for the worker leader

### PR DESCRIPTION
**Pull request by srkukarni**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation
This mr adds an api to check if the worker is ready to serve requests.
### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java [hunk 1]
```diff
@@ -129,4 +129,19 @@ public WorkerInfo getClusterLeader() {
...
+    @GET
+    @ApiOperation(
+            value = "Checks if this node is the leader and is ready to service requests",
+            response = String.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "The requester is not authenticated"),
+            @ApiResponse(code = 503, message = "Worker service is not running")
+    })
+    @Path("/cluster/leaderready")
```
Comment (jerrypeng) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java:142]: I think the url path would be more clean if it is /cluster/leader/ready.  We can add more API under "/leader" in future

Comment (jerrypeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java:142]: Beside we already have have the url path "/cluster/leader"

**Comment by jerrypeng** [comment]
@srkukarni what is the purpose of add this new REST endpoint?

**Comment by srkukarni** [comment]
@jerrypeng the current usecase is for setting up testing pipelines.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java (hunk 1 above)
Comment (srkukarni) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java]: Changed

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java [hunk 2]
```diff
@@ -129,4 +129,19 @@ public WorkerInfo getClusterLeader() {
...
+            value = "Checks if this node is the leader and is ready to service requests",
+            response = String.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "The requester is not authenticated"),
+            @ApiResponse(code = 503, message = "Worker service is not running")
+    })
+    @Path("/cluster/leader/ready")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String isLeaderReady() {
```
Comment (jerrypeng) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java:144]: I think it is more appropriate to just return 200 for OK or 503 (service not available).  That is more idiomatic for HTTP REST calls

Comment (jerrypeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java:144]: Something like
https://github.com/apache/pulsar/blob/master/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/WorkerReadinessResource.java#L65

**Comment by jerrypeng** [comment]
So this endpoint is authentication protected? Is that what we want?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java (hunk 2 above)
Comment (srkukarni) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Worker.java]: Changed

**Pull request merged**